syntax = "proto3";

package bloock;

option go_package = "github.com/bloock/go-bridge/internal/bridge/proto";
option java_package = "bloock.sdk.java.bridge.proto";

// The greeting service definition.
service Greeter {
    // Sends a greeting
    rpc SayHello (HelloRequest) returns (HelloResponse);
    // Sends another greeting
    rpc SayHelloWithError (HelloRequest) returns (HelloResponse);
}

message HelloRequest {
    string name = 1;
}

message HelloResponse {
    string message = 1;
    optional Error error = 2;
}

message Error {
    string kind = 1;
    string message = 2;
}
